tests/activity.go

```diff
@@ -1128,3 +1131,112 @@ func (s *FunctionalSuite) TestActivityHeartBeat_RecordIdentity() {
 	s.NoError(err)
 	s.True(workflowComplete)
 }
+
+func (s *FunctionalSuite) TestActivityTaskCompleteForceCompletion() {
+	sdkClient, err := sdkclient.Dial(sdkclient.Options{
+		HostPort:  s.testCluster.GetHost().FrontendGRPCAddress(),
+		Namespace: s.namespace,
+	})
+	s.NoError(err)
+
+	activityInfo := make(chan activity.Info, 1)
+	taskQueue := s.randomizeStr(s.T().Name())
+	w, wf := s.mockWorkflowWithErrorActivity(activityInfo, sdkClient, taskQueue)
+	s.NoError(w.Start())
+	defer w.Stop()
+
+	ctx := NewContext()
+	workflowOptions := sdkclient.StartWorkflowOptions{
+		ID:        uuid.New(),
+		TaskQueue: taskQueue,
+	}
+	run, err := sdkClient.ExecuteWorkflow(ctx, workflowOptions, wf)
+	s.NoError(err)
+	ai := <-activityInfo
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := sdkClient.DescribeWorkflowExecution(ctx, run.GetID(), run.GetRunID())
+		assert.NoError(t, err)
+		assert.Equal(t, 1, len(description.PendingActivities))
+		assert.Equal(t, "mock error of an activity", description.PendingActivities[0].LastFailure.Message)
+	},
+		10*time.Second,
+		500*time.Millisecond)
+
+	err = sdkClient.CompleteActivityByID(ctx, s.namespace, run.GetID(), run.GetRunID(), ai.ActivityID, nil, nil)
+	s.NoError(err)
+
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := sdkClient.DescribeWorkflowExecution(ctx, run.GetID(), run.GetRunID())
+		assert.NoError(t, err)
+		assert.Equal(t, 0, len(description.PendingActivities))
+		assert.Equal(t, enumspb.WORKFLOW_EXECUTION_STATUS_COMPLETED, description.GetWorkflowExecutionInfo().Status)
+	},
+		10*time.Second,
+		500*time.Millisecond)
+}
```

Comment:
You could wait for the `run` to complete instead here ().

```suggestion
// Ensure the activity is completed and the workflow is unblcked.
s.NoError(run.Get(ctx, nil))
}
```